### PR TITLE
[KOGITO-9311] Suggested improvements for the Kogito Serverless Operator Documentation

### PR DESCRIPTION
**Pull request by robpblake**
**JIRA:** https://issues.redhat.com/browse/KOGITO-9311

**Description:**

This PR provides some suggestions to improve the first-time user experience for the Kogito Serverless Operator documentation. 

I've approached this from the status of someone with no experience of the operator, trying to make things as clear as possible.

I've also tried to make sure we're being as consistent as possible across all of the docs.

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category
- [x] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc [hunk 1]
```diff
@@ -32,6 +32,7 @@ metadata:
     sw.kogito.kie.org/profile: dev <1>
 spec: <2>
   flow:
+    specVersion: 0.8.0
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc:35]: This is a default value on the k8s api. Unfortunately some old versions of the server and kubectl won't replace it with the default value. We need to add a note here about this fix on newer kubectl versions. It's on my plate.

**Review by ricardozanini — APPROVED** [review]
Great PR, Rob! Many thanks for your contribution and trying the operator. The guides look much better now.

**Review by davidesalerno — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/workflow-status-conditions.adoc [hunk 2]
```diff
@@ -105,41 +113,41 @@ Deploying the workflow in xref:cloud/operator/build-and-deploy-workflows.adoc[Pr
...
 | Running
 | False
 | AttemptToRedeployFailed
-| If the workflow deployment is not available, the operator will try to roll out the deployment three times before entering this stage. Check the message in this condition and the workflow pod logs for more info
+| If the Workflow Deployment is not available, the operator will try to roll out the Deployment three times before entering this stage. Check the message in this Condition and the Workflow Pod logs for more info
  
 |===
 
-The normal conditions for the workflow resource are to place a `KogitoServerlessBuild` to run and wait for it to finish. As soon as the image is ready, the workflow transitions to the deployment phase, which is to provision a new workflow service pod to run with the built image.
+When building an image for a Workflow resource, a `KogitoServerlessBuild` to will first be created and the operator will wait for the associated build to finish. Once the build is complete, the Workflow transitions to the deployment phase, which will provision a new Workflow service Pod to run with the built image.
```
[the remark this answers is not in the document]
Comment (robpblake) [reply serverlessworkflow/modules/ROOT/pages/cloud/operator/workflow-status-conditions.adoc:150]: @ricardozanini Good spot. Let me fix that.

**Review by spolti — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/workflow-status-conditions.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (robpblake) [reply serverlessworkflow/modules/ROOT/pages/cloud/operator/workflow-status-conditions.adoc]: Fixed.

**Comment by ricardozanini** [comment]
@kaldesai @radtriste do you mind reviewing this PR?

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/install-serverless-operator.adoc [hunk 3]
```diff
@@ -79,9 +79,9 @@ You can also follow the operator’s log:
 kubectl logs deployment/kogito-serverless-operator-controller-manager -n kogito-serverless-operator-system -f
 ----
 
-Once the operator is running, it will watch for new custom resources (CR) so that you can prepare your environment to be ready to create a new {product_name} application based on the definitions you will send to the operator.
+Once the operator is running, it will watch for instances of the {product_name} Custom Resources (CR). Using CRs, you can configure your {product_name} environment and define Workflows and builds to be handled by the operator.
 
-To check if the definitions are correclty installed, try running:
+For the operator to correctly manage the CR instances, you must first install the {product_name} Custom Resource Definitions (CRDs). To check if the CRD definitions are correctly installed, try running:
```
Comment (MarianMacik) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/install-serverless-operator.adoc:84]:
```suggestion
For the operator to correctly manage the CR instances, you must first install the {product_name} Custom Resource Definitions (CRDs). To check if the CRDs are correctly installed, try running:
```

**Review by MarianMacik — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc [hunk 4]
```diff
@@ -4,9 +4,9 @@
 :description: Developing workflow service with the operator on Kubernetes
 :keywords: kogito, workflow, serverless, operator, kubernetes, minikube, devmode
 
-This document describes how you can develop your workflows directly on Kubernetes with the {operator_name}.
+This document describes how you can develop your Workflows directly on Kubernetes with the {operator_name}.
 
-Workflows in development profile are not tailored for production environments. To build and deploy an immutable workflow application with the operator, see xref:cloud/operator/build-and-deploy-workflows.adoc[].
+Workflows in development profile are not tailored for production environments. To build and deploy an immutable Workflow application with the operator, see xref:cloud/operator/build-and-deploy-workflows.adoc[].
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc:9]:
```suggestion
Workflows in the development profile are not tailored for production environments. To build and deploy an immutable Workflow application with the operator, see xref:cloud/operator/build-and-deploy-workflows.adoc[].
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc [hunk 5]
```diff
@@ -219,43 +222,43 @@ It can give you a clue about what might be happening. See xref:cloud/operator/wo
...
-If you are looking for opening an issue or ask in {product_name} communication channels, this logging information is always useful for the person who will try to help you.
+If you decide to open an issue or ask for help in {product_name} communication channels, this logging information is always useful for the person who will try to help you.
 
 === Possible Failure Scenarios
 
 ==== Feature Not Yet Supported 
 The {operator_name} is under active development. Sometimes a feature might not be available yet. Please see xref:cloud/operator/known-issues.adoc[] for a comprehensive list of available features.
 
-If you identify you're refering to a feature not yet available, please file a new issue so we can prioritize it for you or ask in {product_name} communication channels.
+If you identify you're attempting to use a feature not yet available, please file a new issue, so we can prioritize it for you. Alternatively, please ask in {product_name} communication channels.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc:232]:
```suggestion
If you identify that you are attempting to use a feature that is not available yet, you can file a new issue, so we can prioritize it for you. Alternatively, you can ask in {product_name} communication channels.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/install-serverless-operator.adoc [hunk 6]
```diff
@@ -95,19 +95,19 @@ kogitoserverlessworkflows.sw.kogito.kie.org   2023-03-08T18:31:15Z
 == Uninstall the Operator
 // TODO: this is super verbose for now because we don't have OLM/OperatorHub. This procedure should be replaced as soon as we have them. Or it should be renamed to "Uninstalling snapshot/local versions", so crazy users that don't rely on operatorhub/olm can also install/uninstall their instances.
 
-To uninstall the {operator_name}, first you should remove all the object instances managed by it. Then, you can delete every object created during the installation.
+To uninstall the {operator_name}, first you should delete all the {product_name} Custom Resources (CR) instances managed by it. Once the CRs are deleted, you can then delete all resources created during the operator installation.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/install-serverless-operator.adoc:98]:
```suggestion
To uninstall the {operator_name}, first, you must delete all the {product_name} Custom Resources (CR) instances managed by it. Once the CRs are deleted, you can delete all resources created during the operator installation.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/workflow-status-conditions.adoc [hunk 7]
```diff
@@ -105,41 +113,41 @@ Deploying the workflow in xref:cloud/operator/build-and-deploy-workflows.adoc[Pr
...
 | Running
 | False
 | AttemptToRedeployFailed
-| If the workflow deployment is not available, the operator will try to roll out the deployment three times before entering this stage. Check the message in this condition and the workflow pod logs for more info
+| If the Workflow Deployment is not available, the operator will try to roll out the Deployment three times before entering this stage. Check the message in this Condition and the Workflow Pod logs for more info
  
 |===
 
-The normal conditions for the workflow resource are to place a `KogitoServerlessBuild` to run and wait for it to finish. As soon as the image is ready, the workflow transitions to the deployment phase, which is to provision a new workflow service pod to run with the built image.
+When building an image for a Workflow resource, a `KogitoServerlessBuild` will first be created and the operator will wait for the associated build to finish. Once the build is complete, the Workflow transitions to the deployment phase, which will provision a new Workflow service Pod to run with the built image.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/workflow-status-conditions.adoc:150]:
```suggestion
When building an image for a Workflow resource, a `KogitoServerlessBuild` will first be created and the operator will wait for the associated build to finish. Once the build is complete, the Workflow transitions to the deployment phase, which will provide a new Workflow service Pod to run with the built image.
```

**Review by kaldesai — APPROVED** [review]
@robpblake Very minimal suggestions. Otherwise, your content looks good to me. Thank you!

**Review by kaldesai — APPROVED** [review]
@robpblake Very minimal suggestions. Otherwise, your content looks good to me. Thank you!

**Review by radtriste — APPROVED** [review]

**Pull request merged**